api/nakadi-event-bus-api.yaml

```diff
@@ -1503,11 +1539,40 @@ definitions:
         type: string
         default: 'random'
 
+      compatibility_mode:
+        description: |
+          Compatibility modes provides a mean for event owners to evolve their schema, given changes
+          respect the semantics defined by this field.
+
+          It's designed to be flexible enough so that producers can evolve their schemas while not
+          inadvertently breaking existent consumers.
+
+          Once defined, the compatibility mode is fixed, since otherwise it would break a predefined contract,
+          jeopardizing the value of this feature.
+
+          List of available compatibility modes:
+
+          - 'deprecated': Schema changes are not allowed. This is used only by legacy event types that are not
+            compatible. This option is not available for new event types. Additional properties are allowed.
+
+          - 'incompatible': Any schema modification is accepted, even if it might break existing producers or consumers.
+
+          - 'compatible': Schema changes never break existing consumers. Every event published since the first version
+            is still valid based on the newest schema.
+
+          When in compatible mode, it's allowed to add new optional properties and definitions to an existing schema,
+          but no other changes are allowed.
+        type: string
+        enum:
+          - none
+          - compatible
```

Comment:
Since extending enums is not backward compatible by the api guidelines, maybe we should not explicitly state the enum values. I'm afraid we might add new valued to this enum in the future, when we design a solution for incompatible schema evolution.